nio/client/async_client.py

```diff
@@ -876,6 +876,25 @@ async def login(
 
         return await self._send(LoginResponse, method, path, data)
 
+    def restore_login(
+        self,
+        user_id: str,
+        device_id: str,
+        access_token: str,
+    ):
+        """Restores a previous login to the homeserver.
+
+        Args:
+           user_id (str): The full mxid of the current user.
+           device_id (str, optional): An unique identifier that distinguishes
```

Comment:
Is `device_str` really optional? I think if you're restoring a login with a specific access token you need the ID associated with that token. If you're not sure you should do a quick test and see if it works.

That said, if `device_str` is optional the type in the signature should be `Union[str, None]` or `Optional[str]`.

Reply:
This was a copy-paste error. I've corrected it in my latest commit.